### PR TITLE
Updated matchms_convert tool

### LINKED ISSUE #242: matchms convert: Add different MSP styles as options and improve the tools
The different flavors of MSP files (RIKEN, NIST, matchms, Spectra, etc.) should be interconvertible using matchms. There is current ongoing development for this [issue](https://github.com/matchms/matchms/issues/300) - we should improve there and then implement a simple Galaxy tool which handles this.

Our internal representation of MSP files should likely be the matchms one.

- [ ] Add boolean input for whether to apply metadata harmonization or not
- [ ] Add boolean input on whether to export peak comments or not
- [ ] If `msp` is chosen as output format, expand a new condition choosing the msp flavor with options `matchms`, `NIST`, `RIKEN` and `MassBank`

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR includes the following:

- Added boolean input for whether to apply metadata harmonization
- Added boolean input on whether to export peak comments (for msp)
- Added option to support different export styles for msp
- updated test and testdata
- updated galaxy tool and matchms version (hardcoded for matchms_convert for now)

closes #242

### CODE REVIEW

**Comment by hechth** [comment]
@zargham-ahmad when I converted one of the libraries (Exposome pesticides) to NIST, the retention time field key is not written, but the value is -is there no key defined for retention time in matchms for this style or what happened here?

File: tools/matchms/matchms_convert.xml [hunk 1]
```diff
@@ -16,51 +16,92 @@
...
+                <option value="msp" selected="true">msp</option>
+                <option value="mgf">mgf</option>
+                <option value="json">json</option>
+            </param>
+            <when value="msp">
+                <param label="Export peak comments" name="export_peak_comments" type="boolean"
+                    truevalue="True" falsevalue="False" checked="true"
+                    help="Writes peak comments to individual peaks after the respective mz/intensity pair when set to True. Default is True." />
+                <param label="Export style" name="export_style" type="select"
+                    help="Converts the keys to required Export style. One of ['massbank', 'nist', 'riken', 'gnps']. Default is 'matchms'">
```
Comment (xtrojak) [inline tools/matchms/matchms_convert.xml:48]:
```suggestion
                    help="Converts the keys to required Export style. One of ['matchms', 'massbank', 'nist', 'riken', 'gnps']. Default is 'matchms'">
```

File: tools/matchms/matchms_convert.xml [hunk 2]
```diff
@@ -16,51 +16,92 @@
...
+            <param label="Spectral library format" name="output_format" type="select"
+                help="Output format to convert the spectral library into.">
+                <option value="msp" selected="true">msp</option>
+                <option value="mgf">mgf</option>
+                <option value="json">json</option>
+            </param>
+            <when value="msp">
+                <param label="Export peak comments" name="export_peak_comments" type="boolean"
+                    truevalue="True" falsevalue="False" checked="true"
+                    help="Writes peak comments to individual peaks after the respective mz/intensity pair when set to True. Default is True." />
```
Comment (xtrojak) [inline tools/matchms/matchms_convert.xml:46]: I think in this case, the information about default value is redundant.

```suggestion
                    help="Writes peak comments to individual peaks after the respective mz/intensity pair when selected." />
```

**Review by xtrojak — APPROVED** [review]
Looks good, just some minor comments.

**Comment by zargham-ahmad** [comment]
> @zargham-ahmad when I converted one of the libraries (Exposome pesticides) to NIST, the retention time field key is not written, but the value is -is there no key defined for retention time in matchms for this style or what happened here?

that is because we don't have any entry for retention time for nist but the empty key should not have been written to the file. here is the link to the issue https://github.com/matchms/matchms/issues/422 and the fix for it https://github.com/matchms/matchms/pull/423

**Review by hechth — APPROVED** [review]
Looks good - I will still proceed and merge this and then wait for the next matchms release which will fix this.

**Pull request merged**